File: 251052/Ensign/rest-redis/discussion/src/main/java/com/restApp/discussion/controller/CommentController.java
package com.restApp.discussion.controller;

import com.restApp.discussion.dto.CommentRequestTo;
import com.restApp.discussion.dto.CommentResponseTo;
import com.restApp.discussion.service.CommentService;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1.0/comments")
public class CommentController {

    private final CommentService commentService;

    public CommentController(CommentService commentService) {
        this.commentService = commentService;
    }

    @PostMapping
    public ResponseEntity<CommentResponseTo> create(@RequestBody @Valid CommentRequestTo request) {
        return new ResponseEntity<>(commentService.create(request), HttpStatus.CREATED);
    }

    @GetMapping
    public ResponseEntity<List<CommentResponseTo>> getAll() {
        return ResponseEntity.ok(commentService.findAll());
    }

    @GetMapping("/{id}")
    public ResponseEntity<?> getById(@PathVariable Long id) {
        CommentResponseTo response = commentService.findById(id);
        if (response != null) {
            return ResponseEntity.ok(response);
        }
        return new ResponseEntity<>(
                new com.restApp.discussion.dto.ErrorResponse("Comment not found", "40401"),
                HttpStatus.NOT_FOUND);
    }

    @PutMapping("/{id}")
    public ResponseEntity<?> update(@PathVariable Long id,
            @RequestBody @Valid CommentRequestTo request) {
        CommentResponseTo response = commentService.update(id, request);
        if (response != null) {
            return ResponseEntity.ok(response);
        }
        return new ResponseEntity<>(
                new com.restApp.discussion.dto.ErrorResponse("Comment not found for update", "40402"),
                HttpStatus.NOT_FOUND);
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> delete(@PathVariable Long id) {
        commentService.delete(id);
        return ResponseEntity.noContent().build();
    }

    @GetMapping("/news/{newsId}")
    public ResponseEntity<List<CommentResponseTo>> getByNewsId(@PathVariable Long newsId) {
        return ResponseEntity.ok(commentService.getCommentsByNewsId(newsId));
    }

}

File: 351002/Melnikov/publisher/controllers/comment_controller.py
from fastapi import APIRouter, Response, status

from ..dto.comment_dto import CommentRequestTo, CommentResponseTo
from ..services.comment_service import CommentService

router = APIRouter(prefix="/api/v1.0/comments", tags=["comment"])
service = CommentService()


@router.post("", response_model=CommentResponseTo, status_code=status.HTTP_201_CREATED)
def create_comment(dto: CommentRequestTo):
    return service.create(dto)


@router.get("", response_model=list[CommentResponseTo], status_code=status.HTTP_200_OK)
def get_comments():
    return service.get_all()


@router.get("/{comment_id}", response_model=CommentResponseTo, status_code=status.HTTP_200_OK)
def get_comment(comment_id: int):
    return service.get_by_id(comment_id)


@router.put("/{comment_id}", response_model=CommentResponseTo, status_code=status.HTTP_200_OK)
def update_comment(comment_id: int, dto: CommentRequestTo):
    return service.update(comment_id, dto)


@router.delete("/{comment_id}", status_code=status.HTTP_204_NO_CONTENT)
def delete_comment(comment_id: int):
    service.delete(comment_id)
    return Response(status_code=status.HTTP_204_NO_CONTENT)